meshtastic/tests/test_node_runtime_channel_export.py

```
@@ -0,0 +1,467 @@
+    export_runtime: _NodeChannelExportRuntime,
+    mock_node: MagicMock,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """turn_off_encryption_on_primary_channel logs warning when cache unavailable after write."""
+    primary = _make_channel(
+        0, channel_pb2.Channel.Role.PRIMARY, name="primary", psk=b"\x01"
+    )
+    mock_node.channels = [primary]
+
+    # After write, set channels to None to simulate unavailable cache
+    def write_side_effect(channel):
```

Ruff (0.15.6)

[warning] Missing return type annotation for private function `write_side_effect`

Add return type annotation: `None`

(ANN202)